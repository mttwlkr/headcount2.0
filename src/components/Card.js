import React from 'react';
import '../styles/Card.css';
import PropTypes from 'prop-types';

const Card = ({ info, handleComparison}) => {
  const { location, stats, selected } = info;
  let className = selected ? 'card selected' : 'card';

  const year = Object.keys(stats).map((year, idx) => {
    let className = stats[year] > 0.5 ? 'over-fifty' : 'under-fifty';
    return (  
      <li 
        className={className} 
        key={idx}>
        <span className='year'>{year}</span>
        <span className='card-data'> : {(stats[year] * 100).toFixed(1)}%</span>
      </li> 
    );
  });

  return (
    <article 
      className={className}
      onClick={() => { handleComparison(location); }
      }>
      <h2>{location}</h2>
      <p className='card-label'>Attendance:</p>
      <ul>
        {year}
      </ul>
      <button>{selected ? 'Remove Comparison' : 'Compare District'}</button>
    </article>
  );
};

Card.propTypes = {
  handleComparison: PropTypes.func,
  info: PropTypes.shape({
    location: PropTypes.string,
    stats: PropTypes.object
  }) 
};

export default Card;