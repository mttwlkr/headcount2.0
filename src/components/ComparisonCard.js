import React from 'react'
import PropTypes from 'prop-types'

const ComparisonCard = () => {
  return (
    <div>
    </div>
  )
}

export default ComparisonCard