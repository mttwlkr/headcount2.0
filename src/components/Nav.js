import React, {Component} from 'react';
import '../styles/Nav.css';
import PropTypes from 'prop-types';

class Nav extends Component {
  constructor(props) {
    super(props);
    this.state = {
      search: ''
    }
    this.searchInput = this.searchInput.bind(this)
  }

  searchInput(e) {
    this.props.search(e.target.value)
    this.setState({
      search: e.target.value
    })
  }

  render() {
    return (
      <nav>
        <form>
          <input 
            type='text'
            value={this.state.search}
            onChange={this.searchInput}
          />
        </form>
      </nav>
    )
  }
}

Nav.propTypes = {
  search: PropTypes.func
}

export default Nav;