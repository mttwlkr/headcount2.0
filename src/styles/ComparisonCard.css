.card h3 {
  text-align: center;
  margin-bottom: 0;
}

.card-a,
.card-b {
  height: 100px;
}

.comparison {
  height: 60px;
}

.card .card-a h2,
.card .card-b h2 {
  height: 60px;
  margin-bottom: 0;
}

.comparison-value {
  text-align: center;
  margin: 0;
}