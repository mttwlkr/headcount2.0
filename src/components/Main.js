import React from 'react';
import '../styles/Main.css';
import Card from './Card';
import PropTypes from 'prop-types';

const Main = ( { districts } ) => {
  const district = districts.map((district, idx) => <Card info={district} key={idx}/>);
  return (
    <main className='card-container'>
      {district}
    </main>
  )
}

Main.propTypes = {
  districts: PropTypes.array
}

export default Main