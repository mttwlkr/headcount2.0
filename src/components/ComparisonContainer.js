import React from 'react';
import ComparisonCard from './ComparisonCard'
import Card from './Card'
import PropTypes from 'prop-types';

const ComparisonContainer = ({ cards, handleSelection }) => {

  const [cardA, comparisonCard, cardB] = cards
  
  return (
    <section>
      <Card info={cardA} handleSelection={handleSelection} />
      { comparisonCard && <ComparisonCard comparisonCard={comparisonCard} />}
      { comparisonCard && <Card info={cardB} handleSelection={handleSelection} />}
    </section>
  )
}

ComparisonContainer.propTypes = {
  cards: PropTypes.array
}

export default ComparisonContainer;
