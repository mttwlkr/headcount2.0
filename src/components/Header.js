import React from 'react';
import '../styles/Header.css';
import Nav from './Nav';
import PropTypes from 'prop-types';
import '../styles/Header.css';

const Header = ({search, currentData}) => {
  return (
    <header>
      <div className='header-container'>
        <h1>Welcome to Headcount</h1>
        <Nav search={search}/>
      </div>
    </header>
  )
}

Header.propTypes = {
  search: PropTypes.func
}

export default Header