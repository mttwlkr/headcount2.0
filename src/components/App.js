import React, { Component } from 'react';
import Main from './Main';
import Header from './Header';
import DistrictRepository from '../helpers/helper.js';
import kinderData from '../data/kindergartners_in_full_day_program.js';
import '../styles/App.css';

class App extends Component {
  constructor () {
    super()
    this.state = {
      districtRepository: null,
      districtsArray: null,
      comparisonArray: []
    }
  }

  componentDidMount() {
    this.retrieveSchoolData(kinderData)
  }

  retrieveSchoolData = (schoolData) => {
    const districtRepository = new DistrictRepository(schoolData)
    const districtsArray = districtRepository.findAllMatches()
    this.setState({ districtRepository, districtsArray })
  }

  handleSearch = (userInput) => {
    const userSearch = this.state.districtRepository.findAllMatches(userInput)
    this.setState({
      districtsArray: userSearch
    })
  }

  clearComparisonArray = () => {
    let clearedComparison = [];
    this.updateComparisonArray(clearedComparison)
  }

  handleComparison = (location) => {
    this.updateSelectedFlag(location);
    const {comparisonArray} = this.fetchCurrentState(location);
    const inComparisonArray = comparisonArray.find(district => district.location === location);
    inComparisonArray ? this.removeCardFromComparison(location) : this.addCardToComparison(location); 
  }

  updateSelectedFlag = (location) => {
    const { newRepositoryState, locationToCompare } = this.fetchCurrentState(location);
    locationToCompare.selected = !locationToCompare.selected;
    let newDistrictArray = newRepositoryState.findAllMatches();  
    this.setState({
      districtsArray: newDistrictArray,
      districtRepository: newRepositoryState});
  }

  addCardToComparison = (location) => {
    const {locationToCompare, comparisonArray} = this.fetchCurrentState(location)
    comparisonArray.push(locationToCompare)
    if(comparisonArray.length === 3) {
      let unselectedLocation = comparisonArray.shift();
      this.updateSelectedFlag(unselectedLocation.location);
    }
    this.updateComparisonArray(comparisonArray)
  }

  removeCardFromComparison = (location) => {
    const {comparisonArray } = this.fetchCurrentState(location);
    comparisonArray[0].location === location ? comparisonArray.shift() : comparisonArray.pop();
    this.updateComparisonArray(comparisonArray);
  }

  updateComparisonArray = (comparisonArray) => {
    this.setState({comparisonArray});
  }

  generateComparisons = (location1, location2) => {
    return this.state.districtRepository.compareDistrictAverages(location1, location2)
  }

  fetchCurrentState = (location) => {
    return {
      locationToCompare: this.state.districtsArray.find(district => district.location === location),
      comparisonArray: [...this.state.comparisonArray],
      compareDistrictAverages: this.state.districtRepository.compareDistrictAverages,
      newRepositoryState: Object.assign(new DistrictRepository(kinderData), this.state.districtRepository)
    }
  }

  render() {
    return (
      <div>
        <Header search={this.handleSearch}/>
        {this.state.districtRepository 
          && <Main 
              districts={this.state.districtsArray} 
              handleComparison={this.handleComparison}
              cards={this.state.comparisonArray}
              clearedComparison={this.clearComparisonArray}
              generateComparisons={this.generateComparisons}
              />}
      </div>
    );
  }
}

export default App;
