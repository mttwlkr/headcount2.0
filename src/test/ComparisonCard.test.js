import React from 'react';
import { shallow, mount } from 'enzyme';
import '../components/ComparisonCard'

describe('Comparison Card', () => {

  it('should render without exploding', () => {
    
  })

})
