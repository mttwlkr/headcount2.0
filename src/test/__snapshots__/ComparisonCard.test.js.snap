// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comparison Card should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ComparisonCard
    clearedComparison={[Function]}
    comparisonCard={
        Object {
            "ACADEMY 20": 0.407,
            "COLORADO": 0.53,
            "compared": 1.302,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="card-a"
>
          <h2>
                    COLORADO
                    :
          </h2>
          <p
                    className="comparison-value"
          >
                    Attendance: 
                    53.0
                    %
          </p>
</div>,
        <div
          className="comparison"
>
          <h3>
                    130.2
                    %
          </h3>
          <p
                    className="comparison-value"
          >
                    to
          </p>
</div>,
        <div
          className="card-b"
>
          <h2>
                    ACADEMY 20
                    :
          </h2>
          <p
                    className="comparison-value"
          >
                    Attendance: 
                    40.7
                    %
          </p>
</div>,
        <button
          onClick={[Function]}
>
          Clear Comparison
</button>,
      ],
      "className": "card selected",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              COLORADO
              :
</h2>,
            <p
              className="comparison-value"
>
              Attendance: 
              53.0
              %
</p>,
          ],
          "className": "card-a",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "COLORADO",
                ":",
              ],
            },
            "ref": null,
            "rendered": Array [
              "COLORADO",
              ":",
            ],
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Attendance: ",
                "53.0",
                "%",
              ],
              "className": "comparison-value",
            },
            "ref": null,
            "rendered": Array [
              "Attendance: ",
              "53.0",
              "%",
            ],
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              130.2
              %
</h3>,
            <p
              className="comparison-value"
>
              to
</p>,
          ],
          "className": "comparison",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "130.2",
                "%",
              ],
            },
            "ref": null,
            "rendered": Array [
              "130.2",
              "%",
            ],
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "to",
              "className": "comparison-value",
            },
            "ref": null,
            "rendered": "to",
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              ACADEMY 20
              :
</h2>,
            <p
              className="comparison-value"
>
              Attendance: 
              40.7
              %
</p>,
          ],
          "className": "card-b",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "ACADEMY 20",
                ":",
              ],
            },
            "ref": null,
            "rendered": Array [
              "ACADEMY 20",
              ":",
            ],
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Attendance: ",
                "40.7",
                "%",
              ],
              "className": "comparison-value",
            },
            "ref": null,
            "rendered": Array [
              "Attendance: ",
              "40.7",
              "%",
            ],
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Clear Comparison",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Clear Comparison",
        "type": "button",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="card-a"
>
            <h2>
                        COLORADO
                        :
            </h2>
            <p
                        className="comparison-value"
            >
                        Attendance: 
                        53.0
                        %
            </p>
</div>,
          <div
            className="comparison"
>
            <h3>
                        130.2
                        %
            </h3>
            <p
                        className="comparison-value"
            >
                        to
            </p>
</div>,
          <div
            className="card-b"
>
            <h2>
                        ACADEMY 20
                        :
            </h2>
            <p
                        className="comparison-value"
            >
                        Attendance: 
                        40.7
                        %
            </p>
</div>,
          <button
            onClick={[Function]}
>
            Clear Comparison
</button>,
        ],
        "className": "card selected",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                COLORADO
                :
</h2>,
              <p
                className="comparison-value"
>
                Attendance: 
                53.0
                %
</p>,
            ],
            "className": "card-a",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "COLORADO",
                  ":",
                ],
              },
              "ref": null,
              "rendered": Array [
                "COLORADO",
                ":",
              ],
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Attendance: ",
                  "53.0",
                  "%",
                ],
                "className": "comparison-value",
              },
              "ref": null,
              "rendered": Array [
                "Attendance: ",
                "53.0",
                "%",
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                130.2
                %
</h3>,
              <p
                className="comparison-value"
>
                to
</p>,
            ],
            "className": "comparison",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "130.2",
                  "%",
                ],
              },
              "ref": null,
              "rendered": Array [
                "130.2",
                "%",
              ],
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "to",
                "className": "comparison-value",
              },
              "ref": null,
              "rendered": "to",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                ACADEMY 20
                :
</h2>,
              <p
                className="comparison-value"
>
                Attendance: 
                40.7
                %
</p>,
            ],
            "className": "card-b",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "ACADEMY 20",
                  ":",
                ],
              },
              "ref": null,
              "rendered": Array [
                "ACADEMY 20",
                ":",
              ],
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Attendance: ",
                  "40.7",
                  "%",
                ],
                "className": "comparison-value",
              },
              "ref": null,
              "rendered": Array [
                "Attendance: ",
                "40.7",
                "%",
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Clear Comparison",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Clear Comparison",
          "type": "button",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
